src/tape/ensemble.py

```diff
@@ -1847,6 +1847,36 @@ def _sync_tables(self):
         self.object.set_dirty(False)
         return self
 
+    def select_random_timeseries(self, seed=None):
+        """Selects a random lightcurve from the Ensemble
+
+        Parameters
+        ----------
+        seed: int, or None
+            Sets a seed to return the same object id on successive runs. `None`
+            by default, in which case a seed is not set for the operation.
+
+        Returns
+        -------
+        ts: `TimeSeries`
+            Timeseries for a single object
+
+        """
+
+        if seed is not None:
+            np.random.seed(seed)
+
+        # Avoid a choice from full index space, select a random partition to grab from
+        if self.object.npartitions > 1:
+            partition_num = np.random.randint(0, self.object.npartitions - 1)
+            partition_ids = self.object.get_partition(partition_num).index.values
```

Comment:
What if partition is empty?

Reply:
Wilson brought this up too, it's a good point which I'll need to consider more, might need to have it try and choose again if it's empty.

Reply:
In the latest commit, I took @wilsonbb's suggestion to first shuffle the partitions and then scan through, checking the index lengths until finding a non-empty partition. Added a unit test to cover this case.